sdk/core/azure-core-test/src/main/java/com/azure/core/test/utils/TestProxyUtils.java

```diff
@@ -144,7 +145,12 @@ public static HttpResponse revertUrl(HttpResponse response) {
             } else {
                 currentUrl.setPort(port);
             }
-            response.getRequest().setUrl(currentUrl.toUrl());
+            responseRequest.setUrl(currentUrl.toUrl());
+
+            responseRequest.setHeader(X_RECORDING_UPSTREAM_BASE_URI, null);
+            responseRequest.setHeader(X_RECORDING_MODE, null);
+            responseRequest.setHeader(X_RECORDING_SKIP, null);
+            responseRequest.setHeader(X_RECORDING_ID, null);
```

Comment:
nit: Can we make this `requestResponse.getHeaders().remove(...)`? Semantically the same but clearer than "what does null mean".